### DISCUSSION TITLE
Would I need one or two rasberry pi's for two WaterFurnace systems in the same house

### DISCUSSION THREAD

**Discussion by dkahn93** [ccutrer/waterfurnace_aurora#75] [Q&A]
Hi I am working with a house that has two WaterFurnace HVAC systems in my house (one for each floor, 2 zones on the second floor), each HVAC with an AWL. They are both relatively close in proximity in the basement and it would be great if they could share a rasberry pi. I am setting this up to use on Home Assistant. Would this setup work for on one Rasberry Pi Zero W? 

To recap that would be two USB RS-485 adapters, and two network cables leading to ONE Raspberry Pi Zero W. Would this gem recognize and process both of them without errors?

Or do I need two Rasberry Pi Zero Ws for each WaterFunace system?

Or is there a different set up I'd need in general?

**Comment by ccutrer** [comment]
Yes, it will work fine. You will need two USB adapters, and then you'll just run two instances of the service, each configured to use a different USB adapter. You don't need to worry about keeping them consistent - the startup code will read the serial number from whichever device it's connected to, and use that for the device identifier for Home Assistant.

**Comment by No-Hazmats** [comment]
Hi Cody, I'm not an experienced Linux user and I hope you can help.  Like the the previous questioner, I have two Waterfurnace units.  I have a single Raspberry Pi and it is talking to both USB adapters, and the Fetch commands show it is receiving data from both.  I had put two lines in the single /etc/systemd/system/aurora_mqtt_bridge.service file, one for each USB device.  But only one of the Waterfurnaces shows up in the MQTT Explorer on HA or when I run it from my PC.  So I understand the need for separate services.     I have some questions about how to do that:

1. I had read on other HA discussions about creating multiple instances of a service by adding '%i' into the installation command then creating config files for each instance.  (That example used a .config file, not a .service file.)  Is this the correct way to set up two instances of the bridge service?
2. Is it the /etc/systemd/system/aurora_mqtt_bridge.service file that must be duplicated for each instance of the service?  Are there any other changes required?
3. How does this work with the Mosquitto broker on the HA side of the bridge?  My understanding is that for each instance of the bridge process, the port number needs to change with the single IP address.  I've done some searching but I'm not certain that it can be done with the Mosquitto broker.  Is that possible?  
If not, the options seem to be switching to the EMQX broker, or maybe installing a Mosquitto broker on the Raspi that's connected to my Waterfurnaces and bridging between that and the Mosquitto instance on HA.

**Comment by ccutrer** [comment]
1. No, you should not be adding a `%I` either to the service file name, or inside the service file
2. Yes, you duplicate the service file, giving the file a unique name. Inside the service files, you need to change the command arguments to reference the correct USB device. I would personally figure out the serial of each USB device (in `/dev/serial/by-id/`) and use that rather than the `/dev/ttyUSBX` aliases (which are not guaranteed to come up in the same order on each boot), so that it's consistent which service file is talking to which unit, in case you need to restart just one for some reason. But in reality, it doesn't matter, as long as each service file talks to a different USB device. The service queries whoever it's talking to for the WF ABC board's serial number, and uses that when publishing to MQTT.
3. The bridge process doesn't listen on any sockets, it makes outgoing connections to the MQTT broker, so there's no conflict. You only need one broker, which can accept many connections from many clients (Home Assistant itself and each instance of the bridge service each being clients). Each instance of the bridge service will automatically publish to its own namespace on the broker - `homie/aurora-SERIAL`, where SERIAL is the serial number of the WF ABC board.

**Reply by No-Hazmats** [reply]
Thank you so much.  I have two copies of the service installed.  New Question - This doesn't seem right.  Same result for both copies of the bridge service:

Shep@Aurora:~ $ systemctl status aurora_mqtt_bridge
 aurora_mqtt_bridge.service - Aurora MQTT Bridge
     Loaded: loaded (/etc/systemd/system/aurora_mqtt_bridge.service; enabled; preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Mon 2025-09-22 21:31:40 EDT; 218ms ago
    Process: 2399 ExecStart=/usr/local/bin/aurora_mqtt_bridge dev/ttyUSB1 mqtt://waterfurnace:Lolarosie@192.158.86.51/ (code=exited, status=217/USER)
   Main PID: 2399 (code=exited, status=217/USER)
        CPU: 0

I'm getting this for both services.
This is the content of the .service file (I temporarily switched back to using ttyUSB reference when it seemed to be not working):

[Unit]
Description=Aurora MQTT Bridge

[Service]
User=pi
ExecStart=/usr/local/bin/aurora_mqtt_bridge dev/ttyUSB1 mqtt://waterfurnace:**PW**@192.158.86.51/
Restart=always
RestartSec=3s

[Install]
WantedBy=multi-user.target

**Reply by No-Hazmats** [reply]
Hope you can help.   I have set up two aurora services on the Raspi that's connected to both my waterfurnaces.  The modbus is working on both waterfurnace connections: Aurora_fetch brings back good data from both items and I see unique serial numbers for each.  But only one bridge service is running correctly.  The other fails and is in a restart loop.

### **This bridge is working:**
[Unit]
Description=Aurora MQTT Bridge

[Service]
User=Shep
ExecStart=/usr/local/bin/aurora_mqtt_bridge dev/ttyUSB1 mqtt://waterfurnace:PAssWORD.168.86.121/
Restart=always
RestartSec=3s

[Install]
WantedBy=multi-user.target


### **This Bridge will not run:**
[Unit]
Description=Aurora MQTT Bridge Main House

[Service]
User=Shep
ExecStart=/usr/local/bin/aurora_mqtt_bridge_main_house dev/ttyUSB0 mqtt://waterfurnace:PassWORD@192.168.86.121/
Restart=always
RestartSec=3s

[Install]
WantedBy=multi-user.target

### **This is the result of the Status Request for the failed bridge service:**
Shep@Aurora:~ $ systemctl status aurora_mqtt_bridge_main_house
Aurora_mqtt_bridge_main_house.service - Aurora MQTT Bridge Main House
     Loaded: loaded (/etc/systemd/system/aurora_mqtt_bridge_main_house.service; enabled; preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Tue 2025-10-07 23:10:53 EDT; 2s ago
    Process: 4698 ExecStart=/usr/local/bin/aurora_mqtt_bridge_main_house dev/ttyUSB0 mqtt://waterfurnace:Lolarosie@192.168.86.121/ (code=exited, status=203/EXEC)
   Main PID: 4698 (code=exited, status=203/EXEC)
        CPU: 5ms

**Reply by 0n3man** [reply]
What you've pasted for you service files can't possibly be right.  In both cases the ExecStart line is messed up.  The first parameter in both files should be /usr/local/bin/aurora_mqtt_bridge.  There is no reason you need a second executable named aurora_mqtt_bridge_main_house.  Did you create this second exectuable?  The device name in both cases should start with a /, so /dev/ttyUSB[0 or 1].  The first config is missing @192 in the last parameter.

**Reply by No-Hazmats** [reply]
Well that's the craziest thing.  
First, the missing @192 - I'm afraid that was an artifact of copying & pasting.  I copied multiple copies of the files into notepad and none of them were missing the @192.
I did create the second executable.  See above comment from Cody: since I have two waterfurnaces and need two USB adapters, I need to run two unique instances of the service.

Anyway, I was in the middle of disabling each service one at a time and copying the contents of the service file from the working instance into the non-working instance and so on.  That didn't appear to do any good.  Then I read your comment and went in and added the missing / in front of /dev/ttyUSB...   After I restarted both instances now (finally!) both services are connected to and reporting to HA.  I don't know why the missing slash would allow one instance to work and the other one not.
